dlib/dlibConfig.cmake.in

```diff
@@ -37,3 +37,13 @@ endif()
 
 find_library(dlib_LIBRARIES dlib HINTS ${dlib_INSTALL_PATH}/lib)
 set(dlib_LIBRARIES ${dlib_LIBRARIES} "@dlib_needed_libraries@")
+
+include(CMakeFindDependencyMacro) 
+
+if("@PNG_FOUND@" AND NOT TARGET PNG::png)
```

Comment:
I have checked with davisking and he is ok with putting the hunter changes in the mainline (https://github.com/davisking/dlib/issues/308). But in order to include the changes there is the possibility that huntergate is never called therefore I have these checks to make sure there are no errors for regular builds.